### PR TITLE
(DOCSP-29747): C++: Remove unnecessary page headers, Key Concept verbiage

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-29747

### Staged Changes

- [Staged C++ Docs](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-29747/sdk/cpp/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/cpp/crud/update.txt
Comment (cbullinger) [inline source/sdk/cpp/crud/update.txt:1]: All of the H2s on this page are `Model` -- is that intentional?

File: source/sdk/cpp/sync.txt
Comment (cbullinger) [inline source/sdk/cpp/sync.txt:1]: Should there be an OTP?

File: source/sdk/cpp/sync/sync-subscriptions.txt
Comment (cbullinger) [inline source/sdk/cpp/sync/sync-subscriptions.txt:1]: Out of scope, but should we standardize the prereqs section wording? I'm seeing "Prerequisites" and "Before You Begin"

File: source/sdk/cpp/crud/update.txt
Comment (dacharyc) [reply source/sdk/cpp/crud/update.txt:1]: Ugh, nope, it sure wasn't! I forgot that's why I had this page structured the way it was. Let me do... something... about that.

Thank you for catching this!

**Pull request merged**